### PR TITLE
option to disable scrapy.utils.log.TopLevelFormatter

### PR DESCRIPTION
**Pull request by akhillb**

### CODE REVIEW

**Comment by jdemaeyer** [comment]
Hey @akhillb, thanks for the patch!

What is a use case in which users need to see the full path of an internal Scrapy component?

**Comment by kmike** [comment]
@jdemaeyer see https://github.com/scrapy/scrapy/issues/1309 for some motivation. I wonder if it should be default or not.

**Comment by jdemaeyer** [comment]
Aah I see, I like the educational aspect. I don't think this option will see much use if we don't make it default.

However, the long names will be somewhat in the way during spider development when I just want to look at the requests/scraped items.

**Comment by jdemaeyer** [comment]
Although never mind. The lines are already long due to the URLs so making the component a little longer isn't really "in the way". I like having the long names as default.

File: docs/topics/logging.rst [hunk 1]
```diff
@@ -170,6 +171,10 @@ listed in `logging's logrecord attributes docs
 <https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior>`_
 respectively.
 
+If :setting:`LOG_TOPLEVEL` is set, then the logs will not display the component
+that prints the log. It is unset by default, hence logs contain the component
```
Comment (kmike) [inline docs/topics/logging.rst]: This should be only true for Scrapy components, right? I think it'd be good to mention that.

**Comment by kmike** [comment]
Could you please add a test case?

**Comment by jdemaeyer** [comment]
Really small nitpick ;) How about naming the option `LOG_TOPLEVEL_ONLY`, since the top level is logged in any case?

**Comment by kmike** [comment]
Yeah, name is a bit tricky - LOG_TOPLEVEL is not perfect, and LOG_TOPLEVEL_ONLY may imply that some messages are not logged at all. LOG_SHORT_NAMES?

**Comment by akhillb** [comment]
Sorry for the delay. @kmike i think LOG_SHORT_NAMES is good. @jdemaeyer what do you think about it?

**Comment by jdemaeyer** [comment]
:+1:

**Comment by akhillb** [comment]
Any status regarding this PR?

**Comment by jdemaeyer** [comment]
@akhillb A test would be good

**Comment by kmike** [comment]
After going back and forth I think it is fine to merge it without tests. @akhillb could you please squash the commits?

**Comment by eliasdorneles** [comment]
This PR looks good to me, but it appears that the original branch/repo doesn't exist anymore (Github says `unknown_repository` for the source branch).
@akhillb are you still interested in wrapping this up? Thanks!

**Comment by kmike** [comment]
Continued in https://github.com/scrapy/scrapy/pull/1731.

**Pull request closed without merging**